crates/sui-core/src/event_handler.rs

```diff
@@ -0,0 +1,88 @@
+// Copyright (c) 2022, Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::authority::{ArcWrapper, AuthorityStore};
+use crate::streamer::Streamer;
+use futures::future::join_all;
+use move_bytecode_utils::module_cache::SyncModuleCache;
+use std::sync::Arc;
+use std::time::{SystemTime, UNIX_EPOCH};
+use sui_types::{
+    error::{SuiError, SuiResult},
+    event::{Event, EventEnvelope},
+    messages::TransactionEffects,
+};
+use tokio::sync::mpsc::{self, Sender};
+
+use tracing::{debug, error};
+
+const EVENT_DISPATCH_BUFFER_SIZE: usize = 1000;
+
+pub fn get_unixtime_ms() -> u128 {
+    SystemTime::now()
```

Comment:
We want the UTC timestamp, and 64 bits is sufficient, we don't need 128.  

Instead of the above I advise using `Utc::now().timestamp()` from `use chrono::prelude::*`
(chrono is already a dependency of multiple packages)